contrib/ci-horizen/scripts/test/clang-tidy-launcher.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Get the current commit hash
+export COMMIT=`git log -1 --format="%H"`
```

Comment:
Some considerations can be made on extracting all the commits with respect to remote branch and then iterating the clan-tidy analysis on ech of them; for sure this would require more time for the analysis but the result would be more appropriate, targeting all the commits and not just the last one.

Reply:
Yes, you are right.
However, keeping track of which files have been modified in all the commits is not very convenient and may break in case of force push. Instead, I am experimenting with `run-clang-tidy.py` [works in parallel] to analyze all the codebase, and modify this script accordingly.

Reply:
The whole section has been rewritten to take into account the history of non-analyzed commits. Now the script `clang-tidy_getFileList.py` scans the branch history and considers all the code files in the commits between the last one submitted and the last one analyzed by Codacy in the branch history.
Also, clang-tidy now runs in parallel to further reduce analysis time.